src/main/java/graphql/schema/idl/SchemaTypeExtensionsChecker.java

```diff
@@ -81,18 +83,20 @@ private void checkObjectTypeExtensions(List<GraphQLError> errors, TypeDefinition
                                         checkNamedUniqueness(errors, directive.getArguments(), Argument::getName,
                                                 (argumentName, argument) -> new NonUniqueArgumentError(extension, fld, argumentName))));
 
-                                //
-                                // fields must be unique within a type extension
-                                forEachBut(extension, extensions,
-                                        otherTypeExt -> checkForFieldRedefinition(errors, otherTypeExt, otherTypeExt.getFieldDefinitions(), fieldDefinitions));
-
-                                //
```

Comment:
This code here is inside a loop already - so

* for every type extentions
    * do this check where we loop every type extension

This created a N*M loop